builtin/credential/approle/path_role.go

```diff
@@ -978,6 +1064,7 @@ func (b *backend) pathRoleRead(ctx context.Context, req *logical.Request, data *
 	if err := validateRoleConstraints(role); err != nil {
 		resp.AddWarning("Role does not have any constraints set on it. Updates to this role will require a constraint to be set")
 	}
+	resp.AddWarning(`The "bound_cidr_list" parameter is deprecated and will be removed. Please use "secret_id_bound_cidrs" instead.`)
```

Comment:
Shouldn't this have an if condition so it's only added if they actually set `bound_cidr_list`? Probably this should go up above where you're actually looking at the parameters (although it would rather negate the point of using GetFirst since you'd have to check if bound_cidr_list is set anyways).

Reply:
That would be a somewhat significant code change to make because, if they're using `bound_cidr_list`, the first time the role is read by the `roleEntry` method, if they're using that field it gets moved and cleared. So we'd only know on first read, and it would take some refactoring to move that information out of the `roleEntry` method. 

Another options would be to strip the warning. Or we could leave it there for all read requests during the deprecation period.

Reply:
I think just remove it. I'd rather not return a warning if they didn't actually set it. Chances are that code to handle the old parameter will be there forever anyways :-)

Reply:
After checking out the branch and looking more at the code (outside of just a diff context) it was more clear why this warning was here. I changed the wording slightly, and I think we can leave it in for a bit and then remove both it and the return value.

Reply:
Thank you!

Reply:
Thank _you_ for carrying this through!